src/panels/media-browser/ha-panel-media-browser.ts

```diff
@@ -240,36 +265,57 @@ class PanelMediaBrowser extends LitElement {
   }
 
   private async _startUpload() {
+    if (this._uploading > 0) {
+      return;
+    }
     const input = document.createElement("input");
     input.type = "file";
     input.accept = "audio/*,video/*,image/*";
-    input.addEventListener("change", async () => {
-      try {
-        await uploadLocalMedia(
-          this.hass,
-          this._currentItem!.media_content_id!,
-          input.files![0]
-        );
-      } catch (err: any) {
-        showAlertDialog(this, {
-          text: this.hass.localize(
-            "ui.components.media-browser.file_management.upload_failed",
-            {
-              reason: err.message || err,
+    input.multiple = true;
+    input.addEventListener(
+      "change",
+      async () => {
+        const files = input.files!;
+        const target = this._currentItem!.media_content_id!;
+
+        try {
```

Comment:
What will this try catch, as the try inside the loop will catch everything?

Reply:
I use it for the `finally`.

Reply:
The finally isn't needed then right? If it will not throw anyway?

Reply:
if an error happens in the inner try, we do a `return`.

So now in both error case + when finished will we set uploading to 0 and refresh.

Reply:
Why `return` and not `break`?

Reply:
I don't know 😄

Reply:
Fixed